homeassistant/components/remote/lg_netcast.py

```diff
@@ -0,0 +1,119 @@
+"""
+Remote control support for the LG Netcast TV.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/remote.lg_netcast/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+from requests import RequestException
+import voluptuous as vol
+
+from homeassistant import util
+from homeassistant.components.remote import (RemoteDevice, PLATFORM_SCHEMA,
+                                             ATTR_NUM_REPEATS, ATTR_DELAY_SECS,
+                                             DEFAULT_DELAY_SECS)
+from homeassistant.const import (CONF_HOST, CONF_NAME,
+                                 CONF_ACCESS_TOKEN,
+                                 STATE_OFF, STATE_ON)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pylgnetcast-homeassistant==0.2.0.dev0']
+
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+DEFAULT_NAME = 'LG TV Remote'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_ACCESS_TOKEN):
+        vol.All(cv.string, vol.Length(max=6)),
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the LG Netcast remotes."""
+    from pylgnetcast import LgNetCastClient
+    host = config[CONF_HOST]
+    access_token = config.get(CONF_ACCESS_TOKEN)
+    name = config[CONF_NAME]
+
+    client = LgNetCastClient(host, access_token)
```

Comment:
What happens if the access token is bad? It would be good to validate that before adding the entity.

Reply:
if the access token is bad the entity will be created but it will just not work.
And we can't validate it because the TV could be off when HA starts.
Be aware  this code is the same used by media_player.lg_netcast

Reply:
Isn't it better to try and validate the token, and log an error if we fail and then not add the entit. We can document that the TV should be on when adding the entity. Why would we want to add an entity that doesn't work?

Reply:
Ok, it's a hassle to turn on the TV if there are frequent restarts of home assistant.